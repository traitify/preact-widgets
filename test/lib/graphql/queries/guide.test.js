import guideQuery from "lib/graphql/queries/guide";

describe("guide query", () => {
  it("sets default params", () => {
    const defaultQuery = guideQuery({params: {assessmentId: "xyz", localeKey: "es-US"}});

    expect(defaultQuery).toBe(`{ guide(assessmentId:"xyz",localeKey:"es-US") { competencies { id name introduction order questionSequences { id name personality_type_id questions { id text adaptability order purpose } } } }}`);
  });

  it("overrides default params", () => {
    const fields = [
      "name",
      {competencies: ["id", "name"]}
    ];

    const query = guideQuery({params: {assessmentId: "xyz", localeKey: "es-US"}, fields});

    expect(query).toBe(`{ guide(assessmentId:"xyz",localeKey:"es-US") { name competencies { id name } }}`);
  });
});
