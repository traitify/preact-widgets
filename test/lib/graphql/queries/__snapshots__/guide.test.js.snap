// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graphql Queries Guide get creates query 1`] = `"{ guide(assessmentId:\\"xyz\\",localeKey:\\"es-US\\") { name competencies { id name } }}"`;

exports[`Graphql Queries Guide get creates query with defaults 1`] = `"{ guide(assessmentId:\\"xyz\\",localeKey:\\"es-US\\") { competencies { id name introduction order questionSequences { id name personality_type_id questions { id text adaptability order purpose } } } }}"`;
